pkg/controller/deployment/deployment_controller.go

```diff
@@ -410,10 +410,10 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 	d := *obj.(*extensions.Deployment)
-	if !dc.rsStoreSynced() {
-		// Sleep so we give the replica set reflector goroutine a chance to run.
-		time.Sleep(RSStoreSyncedPollPeriod)
-		glog.Infof("Waiting for replica set controller to sync, requeuing deployment %s", d.Name)
+	if !dc.rsStoreSynced() || !dc.podStoreSynced() {
```

Comment:
How about move this to [L399](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/deployment_controller.go#L399), and insert the key into work queue directly?